src/migrations/dapps/1739898312357_trades-mv.ts

```diff
@@ -0,0 +1,259 @@
+            WHEN COUNT(CASE WHEN st.action = 'executed' THEN 1 END)
+                >= (t.checks ->> 'uses')::int                                       THEN 'sold'
+            ELSE 'open'
+        END AS status
+
+    FROM marketplace.trades      AS t
+    JOIN trades_owner_ok         AS ok
+    ON t.id = ok.id
+
+    JOIN (
+        SELECT
+            ta.id,
+            ta.trade_id,
+            ta.contract_address,
+            ta.direction,
+            ta.beneficiary,
+            ta.extra,
+            erc721_asset.token_id,
+            erc20_asset.amount,
+            item.creator,
+            item.available,
+            nft.owner_address      AS nft_owner,
+            nft.category,
+            nft.id                AS nft_id,
+            nft.issued_id         AS issued_id,
+            nft.name             AS nft_name,
+            coalesce(nft.item_blockchain_id::text, item_asset.item_id) AS item_id
+        FROM marketplace.trade_assets AS ta
+        LEFT JOIN marketplace.trade_assets_erc721 AS erc721_asset
+            ON ta.id = erc721_asset.asset_id
+        LEFT JOIN marketplace.trade_assets_erc20 AS erc20_asset
+            ON ta.id = erc20_asset.asset_id
+        LEFT JOIN marketplace.trade_assets_item AS item_asset
+            ON ta.id = item_asset.asset_id
+        LEFT JOIN ${MARKETPLACE_SQUID_SCHEMA}.item AS item
+            ON ta.contract_address = item.collection_id
+            AND item_asset.item_id::numeric = item.blockchain_id
+        LEFT JOIN ${MARKETPLACE_SQUID_SCHEMA}.nft AS nft
+            ON ta.contract_address = nft.contract_address
+            AND erc721_asset.token_id::numeric = nft.token_id
+    ) AS av
+    ON t.id = av.trade_id
+
+    LEFT JOIN squid_trades.trade AS st
+    ON st.signature = t.hashed_signature
+
+    LEFT JOIN squid_trades.signature_index AS si_signer
+    ON LOWER(si_signer.address) = LOWER(t.signer)
+
+    LEFT JOIN (
+        SELECT *
+        FROM squid_trades.signature_index idx
+        WHERE LOWER(idx.address) IN (
+            '${marketplacePolygon.address}',
+            '${marketplaceEthereum.address}'
+        )
+    ) AS si_contract
+    ON t.network = si_contract.network
+
+    WHERE t.type IN ('public_item_order', 'public_nft_order')
+    GROUP BY
+        t.id,
+        t.type,
+        t.created_at,
+        t.network,
+        t.chain_id,
+        t.signer,
+        t.checks,
+        si_contract.index,
+        si_signer.index;
+    `
+  )
+  pgm.addIndex(materializedViewName, ['id'], { name: 'idx_mv_trades_id', unique: true, ifNotExists: true })
+  pgm.sql(`
+    CREATE OR REPLACE FUNCTION refresh_trades_mv()
+        RETURNS TRIGGER
+        LANGUAGE plpgsql
+        AS $$
+        BEGIN
+        REFRESH MATERIALIZED VIEW CONCURRENTLY ${materializedViewName};
+        RETURN NULL;
+    END;
+    $$;
+  `)
+
+  pgm.sql(`
+    -------------------------------------------------------------------------------
+    -- marketplace.trades
```

Comment:
As the `trades`, `trade_assets`, `trade_assets_erc721` and `trade_assets_erc20` will be changed in a single transaction, is there a way we could avoid refreshing the MV for all of these changes and use only one?